src/pydocstyle.py

```diff
@@ -1400,7 +1413,10 @@ def check_source(self, source, filename):
             for check in self.checks:
                 terminate = False
                 if isinstance(definition, check._check_for):
-                    error = check(None, definition, definition.docstring)
+                    if definition.skips != 'all':
+                        error = check(None, definition, definition.docstring)
+                    else:
+                        error = None
                     errors = error if hasattr(error, '__iter__') else [error]
                     for error in errors:
```

Comment:
You should check if `error` should be skipped _here_, instead of in each individual check function (it will save a lot of boilerplate code below).